noxfile.py

```diff
@@ -147,6 +147,30 @@ def run_pytest(
     )
 
 
+def get_test_files(root: str, ignore_paths: Sequence[str]) -> list[str]:
+    """Returns all test files in the given directory."""
+    paths = []
+
+    for path, _, files in os.walk(root):
+        for name in files:
+            if not any(path.startswith(ignore_path) for ignore_path in ignore_paths):
+                if name.startswith("test_") and name.endswith(".py"):
+                    paths.append(os.path.join(path, name))
```

Comment:
nit: inverting the conditionals can reduce nesting here

```python
if any(path.startswith(ignore_path) for ignore_path in ignore_paths):
    continue
if not name.startswith("test_") or not name.endswith(".py"):
    continue
paths.append(os.path.join(path, name))
```

though I'm undecided about what's better

Reply:
I tend to lean towards avoiding pass/break/continue whenever possible.
Here I think the conditions read much clearer, mainly the second one since we do not have two `not` conditions in the statement.

Reply:
sounds good to me!